### ISSUE TITLE
[Submission] {31} params - {Archie Chaudhury}

### ISSUE THREAD

**Issue by Arch222** [anadim/AdderBoard#40]
### Author

Archie Chaudhury

### Unique Parameter Count

31 (including 9 weight-like hardcoded constants in forward()).

### Accuracy

100%

### Method

Hand-coded weights

### Architecture

1 decoder layer - hidden size = 3 - attention heads = 4 - KV heads = 1 - head dim = 2 - MLP hidden = 4 - vocab size = 10 (digit tokens only)

### Key Tricks

Fixed 3-channel state design: one large constant channel, one digit channel, one tiny decode-slope channel.
Tied embed/decode math: logits come from the same compact digit basis used in embeddings.
RoPE offset-targeted queries: queries are parameterized by 4 learned offsets so each head “locks” onto specific relative positions.
Shared single KV head expanded across 4 Q heads to cut parameters.
Procedural sparse output projection from attention context (only a few nonzero routes are used).
Carry logic encoded in SwiGLU thresholds (piecewise linear separators), not in Python state.
No learned positional embeddings; positional behavior is fixed via sinusoidal/RoPE operations.
Fixed-format prompt and fixed n_digits=10 calibration to keep the constructive weight solution stable.

### Link to Code

https://github.com/Arch222/Addition_Final

### Verification Output

```shell
Model: Handwritten Compressed Transformer Adder
Accuracy: 10010 / 10010 = 100.00%
Time: 93.9s (107 additions/sec)
Status: QUALIFIED (>= 99%)
```

### Additional Notes

Resumbission wih honest

**Comment by anadim** [comment]
Verified! 31 params confirmed (22 nn.Parameter + 9 hardcoded weight constants — honestly declared), 100.00% accuracy (10,010/10,010) reproduced with verify.py (seed=2025).

Audit confirmed:
- Legitimate autoregressive transformer with real GQA attention (4Q heads, 1KV head)
- forward() is standard tensor-in/logits-out — no addition-specific control flow
- The 9 hardcoded constants are sparse projection weights ({-1, 0, +1, ±10}), not algorithmic logic
- Clever RoPE offset-targeted queries for digit routing, SwiGLU thresholds for carry detection
- All addition knowledge is in the weight values, not the code structure

Added to the hand-coded leaderboard at **Rank #2** (tied at 31 params). Thanks for the honest resubmission!
